docs/api.md

```diff
@@ -0,0 +1,531 @@
+- Callback function
+
+#### Example 
+
+See the [ArduinoIoTCloud-Callbacks](https://github.com/arduino-libraries/ArduinoIoTCloud/blob/master/examples/ArduinoIoTCloud-Callbacks/ArduinoIoTCloud-Callbacks.ino) example.
+
+<!-- TCP documentation begins here -->
+
+## ArduinoCloud Class (TCP)
+
+### `begin()` (TCP)
+
+#### Description
+
+Initialises the library with the connection handler specified in `thingProperties.h`. This is automatically generated based on the type of device.
+
+#### Syntax
+
+```
+ArduinoCloud.begin()
+```
+
+#### Parameters
+
+The `ArduinoIoTPreferredConnection` object is created automatically inside a `thingProperties.h` file when configuring a Thing and depends on what type of device you use. The connection handler classes are not part of the Arduino IoT Cloud, but exists inside the [Arduino_ConnectionHandler](https://github.com/arduino-libraries/Arduino_ConnectionHandler) library (listed as a dependency).
+
+For example, using the `WiFiConnectionHandler`, we use the `SSID`, `PASS` parameters which are defined in the `arduino_secrets.h` file, also generated based on what you input in the Arduino IoT Cloud interface.
+
+```
+WiFiConnectionHandler ArduinoIoTPreferredConnection(SSID, PASS);
+```
+
+#### Returns
+Nothing.
+
+### `update()` (TCP)
+
+#### Description
+
+**Note:** for LoRa devices, the `update()` function is implemented differently. See 
+
+Responsible for updating the IoT Cloud connection. 
+- First it feeds a watchdog timer, making sure it is not stuck anywhere,
+- then sends variable updates to the IoT Cloud (via MQTT client)
+- finally, it checks for new data from the IoT Cloud (via the MQTT client)
+
+Any IoT Cloud sketch needs to continuously call the `update()` function, as it will otherwise time out and reset. The `update()` is by default placed inside of the `loop()`.
+
+**Note:** do not use the `delay()` function in the sketch, as the watchdog timer will cause the board to reset and attempt to reconnect.
+
+#### Syntax
+
+```
+ArduinoCloud.update()
+```
+
+#### Parameters
+None.
+
+#### Returns
+Nothing.
+
+
+### `connected()` (TCP)
+
+#### Description
+
+Checks the connection status and returns an integer the connection status (int).
+
+#### Syntax
+
+```
+ArduinoCloud.connected()
+```
+
+#### Parameters
+None.
+
+#### Returns
+- `int` - status of connection
+
+### `printDebugInfo()` (TCP)
+
+#### Description
+
+Print any available debug information.
+
+#### Parameters
+None
+
+#### Returns
+Nothing.
+
+### `setBoardId()`
+
+#### Description
+
+This method is only enabled if you are using an ESP32/ESP8266 board, and sets the board/device id.
+
+#### Syntax
+
+```
+ArduinoCloud.setBoardId(DEVICE_LOGIN_NAME);
+```
+
+### `setSecretDeviceKey()`
+
+#### Description
+
+This method is only enabled if you are using an ESP32/ESP8266 based board, and sets the secret device key for the device. The secret key is only obtainable from the Arduino IoT Cloud during device configuration.
```

Comment:
Temporarily this is also used by the UNO R4, but it will change in the near future.

Reply:
Good call. I added the Nano ESP32 as well.